### PR TITLE
Fix outstanding backup manager issues

### PR DESCRIPTION
**Pull request by BenjaminPelletier**
This PR continues the work in #30 which wasn't quite finished.

The main fix so far is that `--column-statistics=0` is added to the backup script.

### CODE REVIEW

**Comment by BenjaminPelletier** [comment]
Currently having a problem: while trying to more rigorously test database backups, I can backup and restore a full-data site (great).  However, when I try to backup and restore the "empty" database, I'm getting the error ERROR 1146 (42S02) at line 322: Table 'mysql.innodb_index_stats' doesn't exist.  Here are my repro steps:

1. Use this branch
2. Start with everything down (`python deploy_tool.py local_windows dc down` to be sure)
3. Create DB (`python deploy_tool.py local_windows create_db`)
4. Apply "empty" database (`python deploy_tool.py local_windows restore_empty_db`)
5. Start site (`python deploy_tool.py local_windows start_site`)
6. Verify site's up in browser: http://localhost
7. SSH into backup manager on port 22001
    a. `/backup_mysql.sh`
8. Unzip .gz file created in sqlbackup
9. Tear down site (`python deploy_tool.py local_windows dc down`)
10. Create DB second time (`python deploy_tool.py local_windows create_db`)
11. Attempt to restore DB (`python deploy_tool.py local_windows restore_db`)

This is where I get the error.  FWIW, the backup created at 7a is 5.4 MB whereas empty_schema.sql is 1.5 MB (probably generally as expected since the site start adds a bunch of transient state stuff I assume)

**Comment by hcooper** [comment]
I think this might have come about because the `db` container and the `backup_manager` container are running different versions of the mysql tools. (5.7 v's 8.0).

On backup_manager:
```
# mysqldump -V
mysqldump  Ver 8.0.33-0ubuntu0.22.04.2 for Linux on x86_64 ((Ubuntu))

# mysqldump --host ropewiki_db --all-databases --add-drop-database --single-transaction --user=root --password=NotReal
[...]
/*!40000 ALTER TABLE `columns_priv` ENABLE KEYS */;
UNLOCK TABLES;
mysqldump: Couldn't execute 'SELECT COLUMN_NAME,                       JSON_EXTRACT(HISTOGRAM, '$."number-of-buckets-specified"')                FROM information_schema.COLUMN_STATISTICS                WHERE SCHEMA_NAME = 'mysql' AND TABLE_NAME = 'columns_priv';': Unknown table 'COLUMN_STATISTICS' in information_schema (1109)
 ```
 
 On db:
 
 ```
# mysqldump -V
mysqldump  Ver 10.13 Distrib 5.7.42, for Linux (x86_64)

# mysqldump --host ropewiki_db --all-databases --add-drop-database --single-transaction --user=root --password=NotReal
[..... no error.....]
```

It maaaay be better to have them match. But from [what I read](https://serverfault.com/questions/912162/mysqldump-throws-unknown-table-column-statistics-in-information-schema-1109) just adding the `--column-statistics=0` is acceptable too.

**Comment by BenjaminPelletier** [comment]
I think the MySQL version mismatch is probably the likely cause, but I'm not sure what to do about it.  The above procedure still reproduces the problem for me (see below), I don't see an easy way to install mysql-client 5.7, and we'd like to update the server but that's certainly a project for another day.  Prepping the DB by applying [this](https://stackoverflow.com/a/37857010/651139) doesn't immediately fix the issue, but I may be applying it the wrong way.

```
c:\bjp\ropewiki\app>python deploy_tool.py local_windows restore_db
2023-05-23T23:33:46.613240 Restoring backup c:\ropewiki\mount\sqlbackup\all-backup-2023-05-24-063222.sql...
2023-05-23T23:33:46.614240   (NOTE: this operation usually takes a few minutes)
2023-05-23T23:33:46.620242   RUN type c:\ropewiki\mount\sqlbackup\all-backup-2023-05-24-063222.sql | docker container exec -i local_windows_ropewiki_db_1 mysql -uropewiki -pthispasswordonlyworksuntildbisrestored ropewiki
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 322: Table 'mysql.innodb_index_stats' doesn't exist
The process tried to write to a nonexistent pipe.
2023-05-23T23:33:47.395071 Error running command 'type c:\ropewiki\mount\sqlbackup\all-backup-2023-05-24-063222.sql | docker container exec -i local_windows_ropewiki_db_1 mysql -uropewiki -pthispasswordonlyworksuntildbisrestored ropewiki'; rerunning to capture result...
2023-05-23T23:33:47.395071   RUN type c:\ropewiki\mount\sqlbackup\all-backup-2023-05-24-063222.sql | docker container exec -i local_windows_ropewiki_db_1 mysql -uropewiki -pthispasswordonlyworksuntildbisrestored ropewiki
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 322: Table 'mysql.innodb_index_stats' doesn't exist
The process tried to write to a nonexistent pipe.

ERROR: Return code 1
```

**Comment by hcooper** [comment]
I wonder if restoring the DB via the backup_manager makes a difference. (So the same tools were used to both create and restore the db).

e.g. deploy_tool would run something like....

```
cat backup.sql | docker container exec -i dev_ropewiki_backup_manager_1 mysql -h ropewiki_db ...
```

**Comment by BenjaminPelletier** [comment]
I've tested these new changes by successfully doing the following:
1. Perform the reproduction steps above, but observe that restore_db completes successfully
2. Following step 1, make a change to the site, backup again, clear everything, load backup, confirm site change is present
3. Restore the all-backup-2023-05-28-100001.sql production backup locally and confirm that the site looks ok
4. Restore the same backup to dev.rope.wiki along with rsync'd images folder, and visually verify front page looks good

The approach now taken is to just use the same exact base image from the database container for backup manager.  The suggestion to interact with the DB almost exclusively through backup manager is also adopted as that seems like good practice.

File: backup_manager/Dockerfile [hunk 1]
```diff
@@ -3,10 +3,10 @@
 # To build this image, run the following command starting from the root of the repository:
 #   docker image build -f backup_manager/Dockerfile -t ropewiki/backup_manager .
 
-FROM ubuntu:22.04
+FROM mysql:5.7-debian
```
Comment (hcooper) [inline backup_manager/Dockerfile:6]: Using the same basic image for the db and backup manager makes total sense. Everything else LGTM.

**Review by hcooper — APPROVED** [review]

**Pull request merged**